commons/com.b2international.index.lucene/src/com/b2international/index/admin/BaseLuceneIndexAdmin.java

```diff
@@ -210,6 +218,23 @@ public IndexSearcher newSearcher(IndexReader reader, IndexReader previousReader)
 	protected abstract TransactionLog createTransactionlog(Map<String, String> commitData) throws IOException;
 
 	protected abstract Directory openDirectory() throws IOException;
+	
+	@Override
+	public Script compile(String script) {
+		final Class<? extends Script> scriptClass = getScriptClass(script);
+		try {
+			return scriptClass.newInstance();
+		} catch (InstantiationException | IllegalAccessException e) {
+			throw new IndexException("Couldn't instantiate groovy script", e);
+		}
+	}
+	
+	private Class<? extends Script> getScriptClass(String script) {
+		if (!scriptCache.containsKey(script)) {
```

Comment:
I'd still love to see `LoadingCache` here, otherwise two threads can find that the key is not present in the map.